### PR TITLE
Reduced default `ToggleSwitch`'s height to 24px

### PR DESCRIPTION
**Pull request by smmr-dn**
## Changes

Originally, the `ToggleSwitch` checkmark icon of class `iui-toggle-switch-icon` had the margin setup of `iui-size-m` (16px; 8px each) with an additional of 1px on each side (`$iui-toggle-switch-border-thickness`). 

On top of that, the content class `iui-toggle-switch`'s `block-size` was specified by the addition of `--_iui-toggle-switch-handle-size` (16px) + `--_iui-toggle-switch-handle-offset` * 2 (8px) + `$iui-toggle-switch-border-thickness * 2` (2px) which resulted in 26px. 

Therefore, the `block-size` has been reduced in this PR to reduce the height from 26px to 24px while maintaining the icons' size to be 16px for medium (default size).

## Testing

Screenshots containing `ToggleSwitch` with this change have been approved.

## Docs

Added two changesets.

### CODE REVIEW

**Review by mayank99 — DISMISSED** [review]
This has again changed the height of the regular size `ToggleSwitch`. Only the small one should be affected.

**Edit**: Actually it's the opposite: only the regular one should be affected. See [later comment](https://github.com/iTwin/iTwinUI/pull/2363#issuecomment-2532514352).

**Comment by smmr-dn** [comment]
> This has again changed the height of the regular size `ToggleSwitch`. Only the small one should be affected.

I just double checked and realized that there are two sizes for `ToggleSwitch`: small 18px and medium (default) 26px, which have now been reduced to 16px and 24px. Were these supposed to be 16px and 24px instead of 18px and 26px?

**Comment by mayank99** [comment]
> I just double checked and realized that there are two sizes for `ToggleSwitch`: small 18px and medium (default) 26px, which have now been reduced to 16px and 24px. Were these supposed to be 16px and 24px instead of 18px and 26px?

Oh wow, I'm not sure how I missed this. I thought they were the same height as our other components (i.e. 24px and 36px).

Looking at our Figma library, the heights appear to be 18px and 24px, so let's go with that for now.

I personally think 24px is too short for the default toggle switch, but we can have that conversation in the future (outside this PR).

File: .changeset/tough-readers-reflect.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-css': patch
+---
+
+`iui-toggle-switch` now has the height of 24px.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply .changeset/tough-readers-reflect.md]: I have updated the changesets.

**Comment by smmr-dn** [comment]
The 26px in height for default `ToggleSwitch` was intended in the code. The parent's border adds up to the content inside (46x24) which results in the 48x26 intended size.

**Pull request closed without merging**